python/ray/data/dataset.py

```diff
@@ -1180,6 +1180,11 @@ def filter(
         stateful Ray actors. For more information, see
         :ref:`Stateful Transforms <stateful_transforms>`.
 
+        .. tip::
+           If you can represent your filter as an expression that leverages Arrow
+           Dataset Expression, we will be do highly optimized filtering using native
+           Arrow interfaces.
```

Comment:
```suggestion
           If you use the `expr` parameter with a Python expression string, Ray Data optimizes your filter with native Arrow interfaces.
```

> an expression that leverages Arrow Dataset Expression

I don't think users will know what this means. Might be clearer to just say Python expression.

Also, in accordance with our style guide, use present tense and avoid "we":
* https://developers.google.com/style/tense
* https://developers.google.com/style/person

We enforce the style guide in our user guides but not docstrings.